src/core/load-fragment.c

```diff
@@ -6529,7 +6528,7 @@ int config_parse_log_filter_patterns(
                                           "Regex pattern invalid, ignoring: %s=%s", lvalue, rvalue);
         }
 
-        if (pattern_compile_and_log(pattern, 0, &compiled_pattern) < 0)
+        if (pattern_compile_and_log(pattern, 0, NULL) < 0)
```

Comment:
The `pcre2_code` variable `p` in `pattern_compile_and_log()` should be annotated with `_cleanup_(pcre2_code_freep)` to fix the memory leak introduced here